packages/contracts/contracts/discovery/Project.sol

```diff
@@ -83,9 +90,22 @@ contract Project is IProject, ERC165 {
         override(IProject)
         onlyManager(msg.sender)
     {
+        require(approvedByManager == false, "already approved by manager");
+
         approvedByManager = true;
     }
 
+    /// @inheritdoc IProject
+    function approveByLegal()
+        public
+        override(IProject)
+        onlyManager(msg.sender)
```

Comment:
this one should have a different `onlyLegal` modifier, as well as a separate role on the controller side.

The point is precisely that managers and legal are two different entities